polynote-spark/src/main/scala/polynote/kernel/LocalSparkKernel.scala

```diff
@@ -128,19 +138,24 @@ class LocalSparkKernelFactory extends Kernel.Factory.LocalService {
     sparkJars         = (sparkRuntimeJar :: ScalaCompiler.requiredPolynotePaths).map(f => f.toString -> f) ::: scalaDeps.map { case (_, uri, file) => (uri, file) }
     compiler         <- ScalaCompiler(main.map(_._3), sparkRuntimeJar :: transitive.map(_._3), sparkClasspath, updateSettings)
     classLoader       = compiler.classLoader
-    session          <- startSparkSession(sparkJars, classLoader)
+    credentials      <- loadCredentials(polynoteConfig.credentials)
+    session          <- startSparkSession(sparkJars, classLoader, credentials)
     busyState        <- SignallingRef[Task, KernelBusyState](KernelBusyState(busy = true, alive = true))
     interpreters     <- RefMap.empty[String, Interpreter]
     scalaInterpreter <- interpreters.getOrCreate("scala")(ScalaSparkInterpreter().provideSomeLayer[Blocking](ZLayer.succeed(compiler)))
     interpState      <- Ref[Task].of[State](State.predef(State.Root, State.Root))
     closed           <- Promise.make[Throwable, Unit]
   } yield new LocalSparkKernel(compiler, session, interpState, interpreters, busyState, closed)
 
-  private def startSparkSession(deps: List[(String, File)], classLoader: ClassLoader): RIO[BaseEnv with GlobalEnv with CellEnv, SparkSession] = {
+  private def startSparkSession(deps: List[(String, File)], classLoader: ClassLoader,
+                                directCredentials: List[DirectCredentials]): RIO[BaseEnv with GlobalEnv with CellEnv, SparkSession] = {
 
-    // TODO: config option for using downloaded deps vs. giving the urls
-    //       for now we'll just give Spark the urls to the deps
-    val jars = deps.map(_._1)
+    val passwordProtectedHosts = directCredentials.map(_.host).toSet
+
+    val jars = deps.map { case (url, file) =>
+      if (passwordProtectedHosts.contains(URI.create(url).getHost)) file.getAbsolutePath
```

Comment:
`URI.create` could throw if the URI string isn't correctly formatted, so that should be inside an IO. Also, `file.getAbsolutePath` will do blocking IO, so it should be inside `effectBlocking`.